packages/serialize/types/index.d.ts

```diff
@@ -7,18 +7,21 @@ import * as CSS from 'csstype'
 export { RegisteredCache, SerializedStyles }
 
 export type CSSProperties = CSS.PropertiesFallback<number | string>
-export type CSSPropertiesWithMultiValues = {
+export type CSSPropertiesWithMultiValues<Props = unknown> = {
   [K in keyof CSSProperties]:
     | CSSProperties[K]
     | ReadonlyArray<Extract<CSSProperties[K], string>>
+    | ((props: Props) => number | string)
```

Comment:
This should likely be typed like this:
```suggestion
    | ((props: Props) => CSSProperties[K])
```

Reply:
(potentially with support for arrays but those arrays are likely so rarely used... that perhaps we don't have to worry about it here)